src/EnergyPlus/Pipes.cc

```diff
@@ -31,363 +35,93 @@ namespace Pipes {
-				}
-				LocalPipe( PipeNum ).CheckEquipName = false;
+	Array1D< std::shared_ptr< LocalPipeData > > LocalPipe; // dimension to number of pipes
+
+	std::shared_ptr<PlantComponent> LocalPipeData::pipeFactory( std::string objectName ){
+
+		bool found = false;
+		int NumNums;
+		int NumAlphas;
+		int IOStat;
+		bool ErrorsFound = false;
+		std::shared_ptr<LocalPipeData> thisPipe( new LocalPipeData() );
+
+		// search through adiabatic pipes here
+		std::string const cCurrentModuleObject = "Pipe:Adiabatic";
+		int numWaterPipes = InputProcessor::GetNumObjectsFound( cCurrentModuleObject );
+		for ( int PipeWaterNum = 1; PipeWaterNum <= numWaterPipes; ++PipeWaterNum ) {
+			InputProcessor::GetObjectItem( cCurrentModuleObject, PipeWaterNum, cAlphaArgs, NumAlphas, rNumericArgs, NumNums, IOStat );
+			if ( objectName == cAlphaArgs( 1 ) ) {
+				found = true;
+				thisPipe->name = objectName;
+				thisPipe->compType = TypeOf_Pipe;
+				thisPipe->InletNodeNum = GetOnlySingleNode( cAlphaArgs( 2 ), ErrorsFound, cCurrentModuleObject, cAlphaArgs( 1 ), NodeType_Water, NodeConnectionType_Inlet, 1, ObjectIsNotParent );
+				thisPipe->OutletNodeNum = GetOnlySingleNode( cAlphaArgs( 3 ), ErrorsFound, cCurrentModuleObject, cAlphaArgs( 1 ), NodeType_Water, NodeConnectionType_Outlet, 1, ObjectIsNotParent );
+				TestCompSet( cCurrentModuleObject, cAlphaArgs( 1 ), cAlphaArgs( 2 ), cAlphaArgs( 3 ), "Pipe Nodes" );
 			}
 		}
 
-		if ( LocalPipe( PipeNum ).OneTimeInit ) {
-			FoundOnLoop = 0;
-			errFlag = false;
-			ScanPlantLoopsForObject( LocalPipe( PipeNum ).Name, CompType, LocalPipe( PipeNum ).LoopNum, LocalPipe( PipeNum ).LoopSide, LocalPipe( PipeNum ).BranchIndex, LocalPipe( PipeNum ).CompIndex, _, _, FoundOnLoop, _, _, errFlag );
-			if ( FoundOnLoop == 0 ) {
-				ShowFatalError( "SimPipes: Pipe=\"" + PipeName + "\" not found on a Plant Loop." );
-			}
-			if ( errFlag ) {
-				ShowFatalError( "SimPipes: Program terminated due to previous condition(s)." );
+		std::string const cCurrentModuleObject2 = "Pipe:Adiabatic:Steam";
+		int numSteamPipes = InputProcessor::GetNumObjectsFound( cCurrentModuleObject2 );
+		for ( int PipeSteamNum = 1; PipeSteamNum <= numSteamPipes; ++PipeSteamNum ) {
+			InputProcessor::GetObjectItem( cCurrentModuleObject2, PipeSteamNum, cAlphaArgs, NumAlphas, rNumericArgs, NumNums, IOStat );
+			if ( objectName == cAlphaArgs( 1 ) ) {
+				found = true;
+				thisPipe->name = objectName;
+				thisPipe->compType = TypeOf_PipeSteam;
+				thisPipe->InletNodeNum = GetOnlySingleNode( cAlphaArgs( 2 ), ErrorsFound, cCurrentModuleObject2, cAlphaArgs( 1 ), NodeType_Steam, NodeConnectionType_Inlet, 1, ObjectIsNotParent );
+				thisPipe->OutletNodeNum = GetOnlySingleNode( cAlphaArgs( 3 ), ErrorsFound, cCurrentModuleObject2, cAlphaArgs( 1 ), NodeType_Steam, NodeConnectionType_Outlet, 1, ObjectIsNotParent );
+				TestCompSet( cCurrentModuleObject2, cAlphaArgs( 1 ), cAlphaArgs( 2 ), cAlphaArgs( 3 ), "Pipe Nodes" );
```

Comment:
Excellent find @kbenne, but might hold on briefly as I've fixed these and several other things while working on the pipe heat transfer conversion over the weekend and this morning.  

Feel free to review if you want, it just might change _very_ shortly.